NN_module/nn_interface_CAM.f90

```diff
@@ -292,7 +306,7 @@ subroutine interp_to_sam(p_cam, p_surf_cam, var_cam, var_sam, var_cam_surface)
             elseif (p_norm_cam(i, nz_cam) > p_norm_sam(nrf)) then
                 ! This should not happen as CAM grid extends to higher altitudes than SAM
                 ! TODO This check is run on every iteration - move outside
-                write(*,*) "CAM upper pressure level is lower than that of SAM: Stopping."
+!                 write(*,*) "CAM upper pressure level is lower than that of SAM: Stopping."
```

Comment:
Keeping this warning might make sense, as otherwise it's not clear why this is stopping? Or does it happen all the time?

Maybe address the todo above, and move it outside of the iteration

Reply:
I have reimplemented the warning (not expected to ever occur in reality) but left the restructure as a todo in future in the interests of merging this sooner. https://github.com/m2lines/convection-parameterization-in-CAM/pull/44/commits/b2395819081112daf916e75abcfa9134b014a295